src/nprintml/label/aggregator/pcap.py

```diff
@@ -48,93 +55,113 @@ def normalize_npt(cls, npt_csv, path_input_base=None):
                 f"data result but stream was empty: {npt_csv}"
             )
 
+    @staticmethod
+    @storeresults
+    def flatten_npt(indexed_files):
+        print('Loading nPrints')
+
+        # We need every sample size to be the same, so we'll find the
+        # largest nPrint in all of the files and pad each sample with
+        # empty nPrints.
+        header = usecols = None
+        max_length = file_count = 0
+
+        for (file_count, (file_index, npt_file)) in enumerate(indexed_files, 1):
+            if header is None:
+                header = np.genfromtxt(npt_file, delimiter=',', max_rows=1, dtype=str)
+                usecols = np.arange(1, len(header))  # ignore data index (src_ip)
+                skip_header = 1 if isinstance(npt_file, (str, pathlib.Path)) else 0
+            else:
+                skip_header = 1
```

Comment:
Bear in mind that this branching is just optimization (and perhaps superfluous), such that it only grabs the header once, on the first item.

Reply:
Clever. Instead of branching, why not grab it before the loop?

Reply:
Sure thought the same. That's extra too :shrug: :smile_cat:

Reply:
Had the same thought, and easily gave in :smile_cat: 

Like I said, it's actually in some ways more complicated this way, (more lines of code anyway, particularly to handle a lazy iterator, a file descriptor, _etc._); but, in another way, it's more straightforward and has no performance relationship with the length of the iterator (not that that should've mattered anyway).